dim-testsuite/tests/util.py

```diff
@@ -48,10 +48,11 @@ def assertDictSubset(self, actual, expected, msg=None):
         if not (missing or mismatched):
             return
 
+        msg = ''
         if missing:
-            msg = 'Missing: %s' % ','.join(safe_repr(m) for m in missing)
+            msg += 'Missing: %s' % ','.join(safe_repr(m) for m in missing)
         if mismatched:
-            if msg:
+            if len(msg) > 0:
```

Comment:
We could also do
```
msg = []
if missing:
  msg.append('Missing ...')
if mismatched:
  msg.append('Mismatched ...')
  self.fail(msg.join('; '))
```

But this also works

Reply:
Found another place with pretty much the same function:
https://github.com/1and1/dim/blob/ca19f6f536fa5bf2741fcce941aa3769cd12ecf5/ndcli/tests/__init__.py#L13-L35
Did not see any call to that function that uses its `msg` parameter, guess we could just adjust the functions to default to an empty array and go with your suggestion.

Reply:
Whichever way you like. I'm fine with both ways. It worked for the last couple years ^^